### ISSUE TITLE
Bug: 1.4.1 still does not support global configuration

### ISSUE THREAD

**Issue by shaobeichen** [tal-rofe/cz-vinyl#71]
### What did you do?

![image](https://user-images.githubusercontent.com/18715564/185907530-18180ae6-51a9-4fcd-a22d-6671a0d2b744.png)

git cz works, but doesn't use global config.

### What did you expect to happen?

czvinyl supports global configuration

### What actually happened?

![image](https://user-images.githubusercontent.com/18715564/185907991-e7777e82-8c30-441d-b22c-c8e372ea59b8.png)

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by tal-rofe** [comment]
@shaobeichen 
Could you please share what's the directory on which you store the files?

**Comment by shaobeichen** [comment]
C:\Users\xxxxxxx

**Comment by tal-rofe** [comment]
> C:\Users\xxxxxxx

Will check.

**Comment by tal-rofe** [comment]
@shaobeichen 

It may take a while as I only owes macOS machine. If you'd like to make a PR it will be great

**Comment by shaobeichen** [comment]
![image](https://user-images.githubusercontent.com/18715564/187605239-d0a12311-d33b-4ebe-8f43-e8af4c61f0a4.png)

output

```
# console.log(SEARCH_PLACES)
[
  'package.json',   
  '.czvinylrc',     
  '.czvinylrc.json',
  '.czvinylrc.yaml',
  '.czvinylrc.yml', 
  '.czvinylrc.js',  
  '.czvinylrc.ts',  
  '.czvinylrc.cjs', 
  'czvinyl.config.js',
  'czvinyl.config.ts',
  'czvinyl.config.cjs',
  'C:\\Users\\D1M\\package.json',
  'C:\\Users\\D1M\\.czvinylrc',
  'C:\\Users\\D1M\\.czvinylrc.json',
  'C:\\Users\\D1M\\.czvinylrc.yaml',
  'C:\\Users\\D1M\\czvinyl.config.cjs'
]
```
```
# console.log(result, result1 && result1.filepath)
null, C:\Users\D1M\.czvinylrc
```

**Comment by tal-rofe** [comment]
@shaobeichen 

Regarding the 2nd log, What if you were to log the result itself (`result1`)? You'd get the configured file as you expect?

**Comment by shaobeichen** [comment]
> @shaobeichen
> 
> Regarding the 2nd log, What if you were to log the result itself (`result1`)? You'd get the configured file as you expect?

Yes, result1 can get the file path and content I want, but search cannot; I tried to use mjs file to remove loaders, but the result of search is still null

**Comment by tal-rofe** [comment]
@shaobeichen 
Did you try to replace `SEARCH_PLACES` with an hardcoded value? for example `C:\Users\D1M\.czvinylrc`. Would it work? Need to understand whether issue with `cz-vinyl` or with `cosmiconfig`

**Comment by shaobeichen** [comment]
yes, i tried the hardcoded way and it still doesn't work, I set it as below so it doesn't work
```
[
  'C:/Users/D1M/.czvinylrc',
]
[
  'C:\Users\D1M\.czvinylrc',
]
[
  'C:\\Users\\D1M\\.czvinylrc',
]
[
  'package.json',   
  '.czvinylrc',     
  '.czvinylrc.json',
  '.czvinylrc.yaml',
  '.czvinylrc.yml', 
  '.czvinylrc.js',  
  '.czvinylrc.ts',  
  '.czvinylrc.cjs', 
  'czvinyl.config.js',
  'czvinyl.config.ts',
  'czvinyl.config.cjs',
  'C:/Users/D1M/package.json',
  'C:/Users/D1M/.czvinylrc',
  'C:/Users/D1M/.czvinylrc.json',
  'C:/Users/D1M/.czvinylrc.yaml',
  'C:/Users/D1M/czvinyl.config.cjs'
]
[
  'package.json',   
  '.czvinylrc',     
  '.czvinylrc.json',
  '.czvinylrc.yaml',
  '.czvinylrc.yml', 
  '.czvinylrc.js',  
  '.czvinylrc.ts',  
  '.czvinylrc.cjs', 
  'czvinyl.config.js',
  'czvinyl.config.ts',
  'czvinyl.config.cjs',
  'C:\Users\D1M\package.json',
  'C:\Users\D1M\.czvinylrc',
  'C:\Users\D1M\.czvinylrc.json',
  'C:\Users\D1M\.czvinylrc.yaml',
  'C:\Users\D1M\czvinyl.config.cjs'
]
[
  'package.json',   
  '.czvinylrc',     
  '.czvinylrc.json',
  '.czvinylrc.yaml',
  '.czvinylrc.yml', 
  '.czvinylrc.js',  
  '.czvinylrc.ts',  
  '.czvinylrc.cjs', 
  'czvinyl.config.js',
  'czvinyl.config.ts',
  'czvinyl.config.cjs',
  'C:\\Users\\D1M\\package.json',
  'C:\\Users\\D1M\\.czvinylrc',
  'C:\\Users\\D1M\\.czvinylrc.json',
  'C:\\Users\\D1M\\.czvinylrc.yaml',
  'C:\\Users\\D1M\\czvinyl.config.cjs'
]
```

**Comment by tal-rofe** [comment]
@shaobeichen 

Seems like the 3rd party library has an issue.

**Comment by 13OnTheCode** [comment]
@tal-rofe 

You are correct. I have conducted preliminary tests and it seems that the reason is because macOS has a unified directory structure with a single home directory serving as the user's personal file space. On the other hand, Windows uses a partition-based concept and follows a layered directory structure, organizing files and directories using logical disks such as C, D, E, etc., each having its own root directory. After briefly examining the source code of `cosmiconfig`, it appears that no additional handling for Windows systems has been implemented. It is unclear whether this was intentionally done by the developers or if it was overlooked

Testing process:

I executed `cosmiconfig(...).search(...)` inside `D:\Developer\13OnTheCode\test\index.js` As a result, `cosmiconfig` recursively reached the "home directory" on the D drive, rather than the C drive. Therefore, when the configuration file is created in the root directory of the D drive, it can be found successfully

I just discussed ([https://github.com/Zhengqbbb/cz-git/discussions/132](https://github.com/Zhengqbbb/cz-git/discussions/132))  this issue in `cz-git` and submitted a PR ([https://github.com/Zhengqbbb/cz-git/pull/133](https://github.com/Zhengqbbb/cz-git/pull/133))  because it also has the same problem that prevents Windows users from loading the global configuration correctly.


I hope this will be helpful for `cz-vinyl`

**Comment by shaobeichen** [comment]
It seems that v9 has been resolved，https://github.com/cosmiconfig/cosmiconfig/issues/278

**Comment by tal-rofe** [comment]
> It seems that v9 has been resolved，[cosmiconfig/cosmiconfig#278](https://github.com/cosmiconfig/cosmiconfig/issues/278)

Once `cosmiconfig` will upgrade by Renovate to V9

**Comment by tal-rofe** [comment]
@shaobeichen 
I just created a new release on both GitHub & NPM for version `2.5.0` which uses `cosmiconfig@9` new major. Please test it with the new version and update, so we could finally close this issue.
